cmsl1t/config.py

```diff
@@ -0,0 +1,80 @@
+from __future__ import print_function
+import yaml
+import os
+from datetime import datetime
+
+TODAY = datetime.now().timetuple()
+
+
+def validate_sections(expectedSections, configPart):
+    return sorted(expectedSections) == sorted(configPart.keys())
+
+
+def get_unique_out_dir(outdir=None, revision=1):
+    full_outdir = outdir + "-rev_{rev}".format(rev=revision)
+    if os.path.isdir(full_outdir):
+        return get_unique_out_dir(outdir, revision + 1)
+    return full_outdir
+
+
+class ConfigParser(object):
+    SECTIONS = ['general', 'input', 'analysis', 'output']
+
+    def __init__(self):
+        self.config = {}
+
+    def read(self, input_file):
+        cfg = yaml.load(input_file)
+        cfg['general'] = dict(version=cfg['version'], name=cfg['name'])
+        del cfg['version'], cfg['name']
+
+        if not validate_sections(ConfigParser.SECTIONS, cfg):
+            # TODO: improve
+            raise IOError('Invalid config')
+
+        self.config = cfg
+        self.__fill_output_template()
+
+    def sections(self):
+        return self.config.keys()
+
+    def options(self, section):
+        return self.config[section].keys()
+
+    def get(self, section, option):
```

Comment:
Ok, I'm happy with no config validation at this point, I'm just more concerned how we handle situations where an option is requested that wasn't defined. especially when, even with validation, that's a likely use-case.  But for now, we I'll accept this and I'll open a PR with the changes to the `get()` methods that I'm suggesting so we can discuss it there.